dataactvalidator/config/sqlrules/b18_object_class_program_activity_2.sql

```diff
@@ -0,0 +1,9 @@
+SELECT
+    op.row_number,
+    op.object_class,
+    op.by_direct_reimbursable_fun
+FROM object_class_program_activity AS op
+WHERE op.submission_id = {} AND length(op.object_class) = 4 AND NOT COALESCE(op.by_direct_reimbursable_fun,'') = ''
+AND NOT (SUBSTRING(op.object_class,1,1) = '1'
+AND lower(op.by_direct_reimbursable_fun) = 'd') AND NOT (SUBSTRING(op.object_class,1,1) = '2'
+AND lower(op.by_direct_reimbursable_fun) = 'r')
```

Comment:
So the goal here is to allow either the pairing (1,d) or the pairing (2,r).  The two clauses are basically saying 
NOT (1 and d) AND NOT (2 and r)
so it returns any records that don't match either pattern.  I've rearranged the line breaks to make the groupings a bit clearer:
'AND NOT (SUBSTRING(op.object_class,1,1) = '1' AND lower(op.by_direct_reimbursable_fun) = 'd')'
'AND NOT (SUBSTRING(op.object_class,1,1) = '2' AND lower(op.by_direct_reimbursable_fun) = 'r')'
In theory I could use <> by rewriting this part to be something like 'AND '((SUBSTRING(op.object_class,1,1) = '1' AND lower(op.by_direct_reimbursable_fun) <> 'd') OR (SUBSTRING(op.object_class,1,1) = '2' AND lower(op.by_direct_reimbursable_fun) <> 'r') OR
(SUBSTRING(op.object_class,1,1) NOT IN ['1','2']))'